### PR TITLE
759 - Remove autogenerated thumbnail fix

### PR DESCRIPTION
**Pull request by yungalyx**

### CODE REVIEW

**Comment by yungalyx** [comment]
ezgif com-video-to-gif

File: templates/vue/src/components/node-modal/AppearanceForm.vue [hunk 1]
```diff
@@ -102,9 +102,16 @@ export default {
...
+      if (!addThumbnail) {
+        this.node.imageURL = ""
+      }
+    },
+  },
   created() {
-    this.addThumbnail = this.node.imageURL.length > 0
-    this.addLockedThumbnail = this.node.lockedImageURL.length > 0
+    ;(this.addThumbnail = this.node.imageURL.length > 0),
+      (this.addLockedThumbnail = this.node.lockedImageURL.length > 0)
```
Comment (wynnset) [inline templates/vue/src/components/node-modal/AppearanceForm.vue:114]: Not sure why you have added all these symbols. Do they serve a function?

**Review by wynnset — CHANGES_REQUESTED** [review]

File: templates/vue/src/components/node-modal/AppearanceForm.vue (hunk 1 above)
Comment (yungalyx) [reply templates/vue/src/components/node-modal/AppearanceForm.vue:114]: huh, that's weird, I didn't write those extra bits.. maybe it was linting? I'll look into that.

**Comment by yungalyx** [comment]
@wynnset seems that precommit hooks was linting my code, so i just commited and pushed via 
`git commit --no-verify -m "message"`

**Comment by seanRong** [comment]
Tested and LGTM

**Comment by jaskwong** [comment]
Tested, works great! Still has an extra comma in templates/vue/src/components/node-modal/AppearanceForm.vue line 113

File: templates/vue/src/components/node-modal/AppearanceForm.vue [hunk 2]
```diff
@@ -102,8 +102,15 @@ export default {
       addLockedThumbnail: false,
     }
   },
+  watch: {
+    addThumbnail(addThumbnail) {
+      if (!addThumbnail) {
+        this.node.imageURL = ""
+      }
+    },
+  },
```
Comment (shirleywbi) [inline templates/vue/src/components/node-modal/AppearanceForm.vue:111]: Missing implementation for addLockedThumbnail

Comment (shirleywbi) [reply templates/vue/src/components/node-modal/AppearanceForm.vue:111]: Which may be the reason why I'm getting this issue:
1. External Link https://www.google.com/
2. Remove node appearance
3. Click add thumbnail
	- Link for "Show a different thumbnail when locked" auto-fills. When resaved, no UI changes.

**Comment by seanRong** [comment]
Tested, still not saving after I uncheck "add a thumbnail" for a locked node.

**Comment by yungalyx** [comment]
@seanRong I can't recreate the bug you described.. could i get another review?

**Comment by yungalyx** [comment]
@shirleywbi thank you!

**Comment by shirleywbi** [comment]
Tested. Unable to reproduce bug specified by @seanRong. However, came across an issue with not being able to set just the "Show a different thumbnail when locked" without the "Add a thumbnail". Not sure if that's covered in the scope of this issue.
Steps:
1. Create External Link, Locked Node.
2. Remove thumbnail.
3. Click add thumbnail. Do not include image.
4. Enable show a different thumbnail when locked with image.
Issue: No image appears for the locked node.

**Comment by yungalyx** [comment]
@shirleywbi Thank you for the test! I'm not sure if it's desired to have locked thumbnail without a regular node thumbnail. Seems more like a new feature than a bug to me. @narendrasss Can we get your input? Thanks!

**Comment by seanRong** [comment]
Repro steps (just checked):

1. Add external link node (open in new window, I used http://google.com)
2. Access -> unlock node after Oct/31/2020
3. Submit this node
4. Go back in to appearance tab, uncheck box for "add a thumbnail" under node appearance
5. Submit this node again
6. After submission, thumbnail still appears, if you go to edit again, the add a thumbnail box is still checked

**Comment by yungalyx** [comment]
@seanRong Did you pull latest?
ezgif com-video-to-gif (2)

**Comment by seanRong** [comment]
@yungalyx I did, I also did the exact same steps that you've shown in the GIF. I'm really not sure what the reason would be for such a difference. Am I the only one with this issue?

**Comment by narendrasss** [comment]
@yungalyx @shirleywbi from my understanding you should be able to create a locked thumbnail without needing a regular thumbnail.

File: templates/vue/src/components/TapestryNode.vue [hunk 3]
```diff
@@ -198,7 +198,20 @@ export default {
       const showImages = this.settings.hasOwnProperty("renderImages")
         ? this.settings.renderImages
         : true
-      if (this.node.imageURL && this.node.nodeType !== "grandchild" && showImages) {
+      if (
```
Comment (yungalyx) [inline templates/vue/src/components/TapestryNode.vue:201]: php linter why you do this to me ):<

**Comment by yungalyx** [comment]
thumbnail and lockedthumbnail are now seperate.. i.e, users can add locked thumbnail without a default thumbnail.

**Comment by seanRong** [comment]
Relevant code section

`async setLinkData() {
      if (shouldFetch(this.node.typeData.mediaURL, this.node)) {
        const url = this.node.typeData.mediaURL
        const { data } = await getLinkMetadata(url)

        if (data) {
          this.node.typeData.linkMetadata = data

          if (
            !this.node.imageURL ||
            confirm(
              "Would you like to use the link preview image as the thumbnail image?"
            )
          ) {
            this.node.imageURL = data.image
          }
          if (
            !this.node.lockedImageURL ||
            confirm(
              "Would you like to use the link preview image as the locked thumbnail image?"
            )
          ) {
            this.node.lockedImageURL = data.image
          }
        }`

**Comment by yungalyx** [comment]
@seanRong Hi, new implementation will prompt "Would you like to use link preview image as the (locked/unlocked) thumbnail" regardless of whether there is a prior image already set!

**Comment by seanRong** [comment]
I'm getting the prompt more than once. It's appearing every time I toggle add a thumbnail image on/off. It also overrides any link I put in if I click OK for the prompt. Intentional? Could be a bit confusing for the user.

**Comment by yungalyx** [comment]
@seanRong 
- it should prompt on new external link and changes to external links.
- it won't prompt for turning off/on checkbox. 
- Are you adding thumbnail links before or after the prompt? The thumbnail prompt shouldn't prevent you from changing the thumbnail in the future. i.e, adding a new thumbnail after external link should change appearance.

**Comment by yungalyx** [comment]
@seanRong I also can't reproduce the toggle error you mentioned...but the override bug is fixed!
Also, it should prompt twice (once for thumbnail and another for the locked thumbnail).

**Review by wynnset — APPROVED** [review]

**Pull request merged**